front/migrations/20240916_backfill_ds_in_agent_table_query_configurations.ts

```diff
@@ -0,0 +1,106 @@
+import type { LightWorkspaceType } from "@dust-tt/types";
+import assert from "assert";
+import { Op } from "sequelize";
+
+import { Authenticator } from "@app/lib/auth";
+import { AgentTablesQueryConfigurationTable } from "@app/lib/models/assistant/actions/tables_query";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import { DataSourceViewResource } from "@app/lib/resources/data_source_view_resource";
+import { VaultResource } from "@app/lib/resources/vault_resource";
+import type { Logger } from "@app/logger/logger";
+import { makeScript, runOnAllWorkspaces } from "@app/scripts/helpers";
+
+async function backfillDataSourceIdInAgentTableQueryConfigurationForWorkspace(
+  workspace: LightWorkspaceType,
+  logger: Logger,
+  execute: boolean
+) {
+  const auth = await Authenticator.internalAdminForWorkspace(workspace.sId);
+
+  // List workspace data sources.
+  const dataSources = await DataSourceResource.listByWorkspace(auth);
+
+  logger.info(
+    `Found ${dataSources.length} data sources for workspace(${workspace.sId}).`
+  );
+
+  const globalVault = await VaultResource.fetchWorkspaceGlobalVault(auth);
+
+  // Retrieve data source views for data sources.
+  const dataSourceViews =
+    await DataSourceViewResource.listForDataSourcesInVault(
+      auth,
+      dataSources,
+      globalVault
+    );
+
+  // Count agent tables query configurations that uses those data sources and have no dataSourceIdNew.
+  const agentTablesQueryConfigurationsCount =
+    await AgentTablesQueryConfigurationTable.count({
+      where: {
+        // /!\ `dataSourceId` is the data source's name, not the id.
+        dataSourceId: dataSources.map((ds) => ds.name),
+        dataSourceIdNew: {
+          [Op.is]: null,
+        },
+        // Given that the name isn't unique we need to count per workspace.
+        dataSourceWorkspaceId: workspace.sId,
+      },
+    });
+
+  logger.info(
+    `About to update ${agentTablesQueryConfigurationsCount} agent tables query configurations for workspace(${workspace.sId}).`
+  );
+
+  if (!execute) {
+    return;
+  }
+
+  for (const ds of dataSources) {
+    const dataSourceView = dataSourceViews.find(
+      (dsv) => dsv.dataSourceId === ds.id
+    );
+    assert(
+      dataSourceView,
+      `Data source view not found for data source ${ds.id}.`
+    );
+
+    await AgentTablesQueryConfigurationTable.update(
+      // Upsert both `dataSourceIdNew` and `dataSourceViewId` to ensure consistency.
```

Comment:
I would detect and error instead of erasing?

Reply:
While writing this migration, I realized that we also used the data source name in our initial migration to backfill the data source view for the table queries (see code below). The registry lookup logs don't display anything, which I assume is expected, as all those entries were deleted following last week's incident. I'll add a log, but I'd still fix it since I wouldn't be surprised if we come across some instances.

https://github.com/dust-tt/dust/blob/ab92d9f752ed0cfa22c50e9a3ce05bc0197c3d0c/front/migrations/20240902_backfill_views_in_agent_table_query_configurations.ts#L68-L76